tests/lib/external/snapd-testing-tools/utils/log-parser

```diff
@@ -160,17 +126,21 @@ class Rule:
     """
 
     def __init__(self, rule):
-        parts = rule.split('=',1)
+        parts = rule.split("=", 1)
         if len(parts) != 2:
-            raise ValueError("Error: Rule '{}' does not follow the KEY=PATTERN format".format(rule))
+            raise ValueError(
+                "Error: Rule '{}' does not follow the KEY=PATTERN format".format(rule)
+            )
 
         self.key = parts[0]
         self.pattern = parts[1]
 
         try:
             re.compile(self.pattern)
         except re.error as err:
-            raise ValueError("Error: pattern '{}' cannot be compiled: {}".format(pattern, err))
+            raise ValueError(
+                "Error: pattern '{}' cannot be compiled: {}".format(self.pattern, err)
+            )
 
     def filter(self, lines):
```

Comment:
@zyga wrote: This should re-use a compiled pattern stored as an instance attribute.